salt/utils/pycrypto.py

```diff
@@ -26,7 +26,10 @@
 try:
     import crypt
 
-    HAS_CRYPT = True
+    # We need to ensure if the right `crypt` is loaded,
+    # as LazyLoader can load `salt.utils.crypt` instead of `crypt`
+    # if there is Python has no `crypt` (was removed in 3.11).
```

Comment:
`if Python has no crypt`?

Reply:
I was more talking about the wording in the comment. It's not clear what you're saying here. I was guessing.

Reply:
> In this case there are some chances that instead of crypt from Python LazyLoader will load salt.utils.crypt instead.

Doesn't that just point at a bug in the LazyLoader?
I was shocked when I realised recently that it will import relative packages..
I'm not sure this change is actually a fix, but more a workaround for the actual problem

Reply:
@frebib, you are right, most probably it's a bug in LazyLoader. Will try to check with it, whi it's trying to load `salt.utils.crypt` but not sure how much time it could take.

Reply:
@frebib, @twangboy I finally found the exact root cause of such behaviour of LazyLoader. It was added with this change https://github.com/saltstack/salt/pull/53167 (in brief it's adding `salt.utils` to `sys.path` on loading modules), but for me looks like it was intentional and here we just see a kind of corner case, but strange thing that in general it's quite possible that LazyLoader could accidentally load unexpected modules from salt codebase matching the names of python core ones.
Maybe @aplanas could give us a hand here, but I think this PR still makes sense as most probably we will not get any significant amount of similar cases.